Essentials/src/com/earth2me/essentials/services/ReserveEssentials.java

```diff
@@ -0,0 +1,966 @@
+package com.earth2me.essentials.services;
+
+import com.earth2me.essentials.Essentials;
+import com.earth2me.essentials.User;
+import com.earth2me.essentials.api.NoLoanPermittedException;
+import com.earth2me.essentials.api.UserDoesNotExistException;
+import net.ess3.api.Economy;
+import net.tnemc.core.Reserve;
+import net.tnemc.core.economy.EconomyAPI;
+import net.tnemc.core.economy.currency.Currency;
+import net.tnemc.core.economy.response.AccountResponse;
+import net.tnemc.core.economy.response.EconomyResponse;
+import net.tnemc.core.economy.response.GeneralResponse;
+import net.tnemc.core.economy.response.HoldingsResponse;
+import org.bukkit.Bukkit;
+import org.bukkit.World;
+
+import java.math.BigDecimal;
+import java.util.UUID;
+
+import static com.earth2me.essentials.I18n.tl;
+
+/**
+ * @author creatorfromhell <daniel.viddy@gmail.com>
+ */
+public class ReserveEssentials implements EconomyAPI {
+
+    private final Essentials ess;
+
+    public ReserveEssentials(Essentials plugin) {
+        this.ess = plugin;
+    }
+
+    public static void register(Essentials plugin) {
+        //Check to see if there is an economy provider already registered, if so Essentials will take the back seat.
+        if (!((Reserve) Bukkit.getServer().getPluginManager().getPlugin("Reserve")).economyProvided()) {
+            Reserve.instance().registerProvider(new ReserveEssentials(plugin));
+        }
+    }
+
+    /**
+     * @return The name of the Economy implementation.
+     */
+    @Override
+    public String name() {
+        return "EssentialsX";
+    }
+
+    /**
+     * @return The version of Reserve the Economy implementation supports.
+     */
+    @Override
+    public String version() {
+        return "0.1.4.6";
+    }
+
+    //This is our method to convert UUID -> username for use with Essentials' create account methods.
+    private String getName(UUID identifier) {
+        final User user = ess.getUser(identifier);
+        return ((user == null) ? identifier.toString() : user.getName());
+    }
+
+    /**
+     * @return Whether or not this implementation is enabled.
+     */
+    @Override
+    public boolean enabled() {
+        return true;
+    }
+
+    /**
+     * @return Whether or not this implementation should have a default Vault implementation.
+     */
+    @Override
+    public boolean vault() {
+        return false;
+    }
+
+    /**
+     * Used to get the plural name of the default currency.
+     * @return The plural name of the default currency.
+     */
+    @Override
+    public String currencyDefaultPlural() {
+        return tl("moneyPlural");
```

Comment:
Late catch but this should probably be defined in the config, not the locale. Most other currency formatting settings are in `config.yml`.